local-tests/setup/tinny-environment.ts

```diff
@@ -0,0 +1,371 @@
+
+      if (index !== -1) {
+        // If an available key is found
+        this.processEnvs.KEY_IN_USE[index] = true; // Mark the key as in use
+        console.log('[𐬺🧪 Tinny Environment𐬺] 🔑 Selected key at index', index); // Log a message indicating that we have selected a key
+
+        // Set a timer to automatically release the key after 10 seconds
+        setTimeout(() => {
+          this.releasePrivateKey(index);
+          console.log(
+            '[𐬺🧪 Tinny Environment𐬺] 🔓 Automatically released key at index',
+            index,
+            `after ${this.processEnvs.TIME_TO_RELEASE_KEY / 10000} seconds`
+          );
+        }, this.processEnvs.TIME_TO_RELEASE_KEY);
+
+        return { privateKey: this.processEnvs.PRIVATE_KEYS[index], index }; // Return the key and its index
+      } else {
+        console.log('[𐬺🧪 Tinny Environment𐬺] No available keys. Waiting...'); // Log a message indicating that we are waiting
+        // Wait for the specified interval before checking again
+        await new Promise((resolve) =>
+          setTimeout(resolve, this.processEnvs.WAIT_FOR_KEY_INTERVAL)
+        );
+      }
+    }
+  }
+
+  /**
+   * Marks a private key as available again after use.
+   * @param {number} index - The index of the key to mark as available.
+   */
+  releasePrivateKey(index: number) {
+    this.processEnvs.KEY_IN_USE[index] = false;
+    console.log(
+      `[𐬺🧪 Tinny Environment𐬺] 🪽 Released key at index ${index}. Thank you for your service!`
+    );
+  }
+
+  /**
+   * Initializes the LitNodeClient based on the specified network configuration and environment variables.
+   * This setup differentiates between local and production environments, adjusts node attestation checks,
+   * and sets network-specific parameters. The function ensures the client is connected and ready before proceeding.
+   *
+   * The LitNodeClient is configured differently based on the network:
+   * - LOCALCHAIN: Uses custom settings for local testing, with node attestation disabled.
+   * - MANZANO (or other specified testnets): Configures for specific network environments with node attestation enabled.
+   *
+   * Logs the process and exits if the client is not ready after attempting to connect.
+   */
+
+  async setupLitNodeClient() {
+    console.log('[𐬺🧪 Tinny Environment𐬺] Setting up LitNodeClient');
+
+    if (this.network === LIT_TESTNET.LOCALCHAIN) {
+      this.litNodeClient = new LitNodeClient({
+        litNetwork: 'custom',
+        bootstrapUrls: this.processEnvs.BOOTSTRAP_URLS,
+        rpcUrl: this.processEnvs.LIT_RPC_URL,
+        debug: this.processEnvs.DEBUG,
+        checkNodeAttestation: false, // disable node attestation check for local testing
+        contractContext: networkContext as LitContractContext,
+      });
+    } else if (this.network === LIT_TESTNET.MANZANO) {
+      this.litNodeClient = new LitNodeClient({
+        litNetwork: this.network, // 'habanero' or 'manzano'
```

Comment:
We are not supporting Habanero by default in the tests since it's a public mainnet, and `manzano` shares the same build, but you can add that if you wish.